pytorch_lightning/trainer/training_io.py

```diff
@@ -238,7 +238,9 @@ def dump_checkpoint(self):
         checkpoint['state_dict'] = model.state_dict()
 
         if hasattr(model, "hparams"):
-            checkpoint['hparams'] = vars(model.hparams)
+            is_namespace = isinstance(model.hparams, Namespace)
+            checkpoint['hparams'] = vars(model.hparams) if is_namespace else model.hparams
+            checkpont['hparams_type'] = 'namespaces' if is_namespace else 'dict'
```

Comment:
@awaelchli @Borda fixed it... now need to fix load

Reply:
how does that address your concern about hparams[".." ] access? load_from_checkpoint will pass a Namespace to Lightningmodule.

Reply:
ok nvm :)

Reply:
well I would rather stay with the fixed interface - `hparams` as Namespace property and setter